### PR TITLE
API libs: CJS -> ESM / Node 20

### PR DESCRIPTION
**Pull request by CarsonF**

### CODE REVIEW

File: tools/adjust-package-exports.mts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { readFile, writeFile } from 'node:fs/promises';
-import { PackageJson } from 'type-fest';
```
Comment (bryanjnelson) [inline tools/adjust-package-exports.mts:2]: I know there's a reason, but you say "upgrade"...how does this upgrade?

File: tools/adjust-package-exports.mts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import { resolve } from 'node:path';
 
 (async () => {
   const dir = resolve('dist');
   const pkg = JSON.parse(await readFile(`${dir}/package.json`, 'utf8'));
-  pkg.exports = mapValues(pkg.exports, (exp: Record<PackageJson.ExportCondition, string>) => ({
+  pkg.exports = mapValues(pkg.exports, (exp: Record<string, string>) => ({
     import: exp.import.replace(/\/src\//, '/').replace(/\.ts$/, '.js'),
+    default: exp.default.replace(/\/src\//, '/').replace(/\.ts$/, '.js'),
     require: exp.require.replace(/\/src\//, '/').replace(/\.ts$/, '.cjs'),
```
Comment (bryanjnelson) [inline tools/adjust-package-exports.mts:10]: Still need `require` here?  I'm sure you're not wrong, but I'm curious.

**Review by bryanjnelson — APPROVED** [review]
Forward!

File: tools/adjust-package-exports.mts (hunk 1 above)
Comment (CarsonF) [reply tools/adjust-package-exports.mts:2]: They removed the string literal keys I was using in the latest version. Replaced on our side with `string`, I guess it's a bit more dynamic than expected.

File: tools/adjust-package-exports.mts (hunk 2 above)
Comment (CarsonF) [reply tools/adjust-package-exports.mts:10]: Only handling API libs here ATM, since the UI servers are still CJS.

Comment (CarsonF) [reply tools/adjust-package-exports.mts:10]: This whole setup is wonky & hacky. I think there was an attempt to use src for dependent libraries while here in this monorepo, but then change to reference dist/ when actually published.

**Comment by CarsonF** [comment]
YOLO

**Pull request merged**